libs/pages/services/src/lib/ui/page-settings-deployment-pipeline/draggable-item/draggable-item.tsx

```diff
@@ -0,0 +1,68 @@
+import { CloudProviderEnum, ServiceTypeEnum } from 'qovery-typescript-axios'
+import { DraggableProvided, DraggableStateSnapshot } from 'react-beautiful-dnd'
+import { getServiceType } from '@qovery/shared/enums'
+import { ApplicationEntity, DatabaseEntity } from '@qovery/shared/interfaces'
+import { BadgeService, Truncate } from '@qovery/shared/ui'
+import { upperCaseFirstLetter } from '@qovery/shared/utils'
+
+export interface DraggableItemProps {
+  services: (DatabaseEntity | ApplicationEntity)[]
+  serviceId: string
+  cloudProvider: CloudProviderEnum
+  provided?: DraggableProvided
+  snapshot?: DraggableStateSnapshot
+}
+
+const getServiceByServiceId = (
+  serviceId: string,
+  services: (DatabaseEntity | ApplicationEntity)[]
+): DatabaseEntity | ApplicationEntity => {
+  return services.filter((service) => service.id === serviceId)[0]
+}
+
+export function DraggableItem(props: DraggableItemProps) {
+  const { services, serviceId, cloudProvider, provided, snapshot } = props
+
+  const service = getServiceByServiceId(serviceId || '', services)
+  const serviceType = getServiceType(service)
+
+  const classNameItem = (isDragging: boolean) =>
+    `flex items-center drop-shadow-item-deployment-group bg-element-light-lighter-100 rounded px-2 py-3 border ${
+      isDragging ? 'border-2 border-success-500' : 'border-element-light-lighter-400'
+    }`
+
+  const contentWithParams = serviceType === ServiceTypeEnum.DATABASE
+
+  const content = (name: string, type: string, mode: string) => {
```

Comment:
Why don't you put the content directly in the return part of the component ? Why do you need to extract this part in a function?

Reply:
I took it out to make it easier to read, but I can put it inside if you want